File: Mo-DBRS_API/Matlab/experimental_computer_example.m

%% Initialize
%%
%% Biometrics
biopac_client = biopac_initialize('/dev/tty.usbserial-BBTKUSBTTL')

%% Eye-tracking
pupil_client = pupil_initialize('192.168.1.10', 50020);

%% Mo-DBRS RP Connection
modbrs_client = mo_dbrs_initialize('192.168.1.9', 8080);




%%
%% Do Work
%% ...
%%




%% Send Marks
%%
%% Biometrics
biopac_send_pulse(biopac_client, 0.4)

%% Eye-tracking
pupil_send_mark(pupil_client);

%% Mo-DBRS send Mark
mo_dbrs_send_mark(modbrs_client);




%% Stimulate
mo_dbrs_send_stim(modbrs_client);

%% Store RTE
mo_dbrs_send_store(modbrs_client);

%% Store Magnet iEEG
mo_dbrs_send_magnet(modbrs_client);

File: Mo-DBRS_API/Python/experimental_computer_example.py
import biopac
import modbrs
import pupil
import time

modbrs_ip = 192.168.0.2
modbrs_port = 50000

pupil_ip = 192.168.0.10
pupil_port = 50000

biopac_port = '/dev/USB...'

modbrs_client = modbrs_initialize(modbrs_ip, modbrs_port)
pupil_client, pupil_client_pub = pupil_initialize(pupil_ip, pupil_port)
biopac_client = biopac_initialize(biopac_port)

pupil_start_recording(pupil_client)

time.sleep(5)


modbrs_send_mark(modbrs_client)
biopac_send_pulse(biopac_client, 0.4)
pupil_send_annotation(pupil_client_pub)

time.sleep(5)

pupil_stop_recording(pupil_client)
pupil_client.close()
pupil_cilent_pub.close()
modbrs_client.close()
biopac_client.close()

